src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1104,25 +1104,37 @@ public Socket Accept()
 
             // This may throw ObjectDisposedException.
             SafeSocketHandle acceptedSocketHandle;
-            SocketError errorCode = SocketPal.Accept(
-                _handle,
-                socketAddress.Buffer,
-                ref socketAddress.InternalSize,
-                out acceptedSocketHandle);
+            SocketError errorCode;
+            try
+            {
+                errorCode = SocketPal.Accept(
+                    _handle,
+                    socketAddress.Buffer,
+                    ref socketAddress.InternalSize,
+                    out acceptedSocketHandle);
+            }
+            catch (Exception ex)
```

Comment:
Is it expected to catch here anything else than `ObjectDisposedException`?

Reply:
I don't know. Unless catching the base exception is somehow more expensive than catching `ObjectDisposed`, I would prefer to keep it this way considering we're just rethrowing it anyway.